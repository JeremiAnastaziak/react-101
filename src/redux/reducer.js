import { handleActions, combineActions } from 'redux-actions';
import { increment, decrement } from './actions';

const defaultState = {
    counter: 10
};

export default handleActions({
        [combineActions(increment, decrement)]: (
            state,
            {
                payload: {
                    amount
                }
            }
        ) => {
            return { ...state,
                counter: state.counter + amount
            };
        }
    },
    defaultState
);
