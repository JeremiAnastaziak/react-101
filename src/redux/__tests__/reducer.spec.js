import reducer from '../reducer';

describe('reducer', () => {
  test('handles actions', () => {
    expect(
      reducer(
        undefined,
        {
          type: 'INCREMENT',
          payload: {
            amount: 1,
          },
        },
      ),
    ).toMatchSnapshot();

    expect(
      reducer(
        undefined,
        {
          type: 'DECREMENT',
          payload: {
            amount: -1,
          },
        },
      ),
    ).toMatchSnapshot();
  });

  test('returns default state on unknown action', () => {
    expect(
      reducer(
        undefined,
        {
          type: 'UNKNOWN_ACTION',
          payload: {
            amount: -1,
          },
        },
      ),
    ).toMatchSnapshot();
  });
});
