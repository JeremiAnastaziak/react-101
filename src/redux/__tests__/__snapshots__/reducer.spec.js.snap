// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer handles actions 1`] = `
Object {
  "counter": 11,
}
`;

exports[`reducer handles actions 2`] = `
Object {
  "counter": 9,
}
`;

exports[`reducer returns default state on unknown action 1`] = `
Object {
  "counter": 10,
}
`;
